lib/master.js

```diff
@@ -41,26 +41,58 @@ function isPowerOfTwo(n) {
 	return (n !== 0) && ((n & (n - 1)) === 0);
 }
 
+var ONE_MB = 1024 * 1024;
+
+// for backward compatibility, we check supplied options and normalize it to the new format
+function normalize_options(options) {
+	// if new-style options are supplied, we take them and ignore (=remove) old-style options
+	if ('retire_rss_mb'    in options) delete options.retire_rss;
+	if ('total_maxheap_mb' in options) delete options.totalmaxheap;
+
+	// Thresholds used to be provided in bytes. In 0.0.9, we prefer MB.
+	// For clarity, new projects using ipcluster are expected to use the explicit new properties retire_rss_mb and total_maxheap_mb
+	// However, some projects might be passing retire_rss and totalmaxheap in MB.
+	// The code below normalizes when needed.
+	// It works based on the ^6 order of magnitudes between bytes and MB, which allow us to make 2 assumptions:
+	// if a value is supplied in bytes, it will not be less than ONE_MB (that would be insufficient for practical purposes)
+	// if a value is supplied in MB, it will not be more than ONE_MB (that would allocate 1TB RAM!)
+
+	if ('retire_rss' in options) {
+		options.total_maxheap_mb = options.retire_rss;
+		delete options.retire_rss;
+		if (options.retire_rss_mb > ONE_MB) options.retire_rss_mb /= ONE_MB;
```

Comment:
This code makes it look like we're apply the magic on `retire_rss_mb`. I think it's clearer if we do:
```javascript
if ('retire_rss' in options) {
    if (options.retire_rss > ONE_MB)
        options.retire_rss_mb = options.retire_rss / ONE_MB;
    else
        options.retire_rss_mb = options.retire_rss;

    delete options.retire_rss;
}
```